geode-for-redis/src/test/java/org/apache/geode/redis/internal/data/collections/SizeableByteArrayListTest.java

```diff
@@ -73,22 +74,190 @@ public void removeObjects_getSizeInBytesIsAccurate() {
     assertThat(list.size()).isEqualTo(0);
   }
 
+  @Test
+  public void removeObject_getSizeInBytesIsAccurate() {
```

Comment:
The test above this one, `removeObjects_getSizeInBytesIsAccurate()`, could be improved, since it's currently possible that it will remove all the elements from the list in one go, or never test the case where the count is negative. A better test might be something like:
```
  @Test
  public void removeObjects_getSizeInBytesIsAccurate() {
    // Create a list with only duplicate elements and confirm that it correctly reports its size
    SizeableByteArrayList list = new SizeableByteArrayList();
    byte[] bytes = "anElement".getBytes(StandardCharsets.UTF_8);
    for (int i = 0; i < INITIAL_NUMBER_OF_ELEMENTS; ++i) {
      // Clone the byte array because otherwise we end up with the list containing multiple
      // references to the same object in memory rather than references to multiple different
      // objects
      list.addFirst(bytes.clone());
    }
    assertThat(list.getSizeInBytes()).isEqualTo(sizer.sizeof(list));

    // Remove elements from the head
    list.remove(bytes, INITIAL_NUMBER_OF_ELEMENTS / 4);
    assertThat(list.getSizeInBytes()).isEqualTo(sizer.sizeof(list));

    // Remove elements from the tail
    list.remove(bytes, INITIAL_NUMBER_OF_ELEMENTS / 4);
    assertThat(list.getSizeInBytes()).isEqualTo(sizer.sizeof(list));

    // Remove all of the remaining elements
    list.remove(bytes, 0);
    assertThat(list.getSizeInBytes()).isEqualTo(sizer.sizeof(list));
    assertThat(list).isEmpty();
  }
```

Reply:
Added